### PR TITLE
Ask for missing information

### PR DESCRIPTION
**Pull request by skitt**
If the user hasn’t provided a cluster id or color codes on the command
line, ask for them.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

---
This change is [Reviewable](https://reviewable.io/reviews/submariner-io/submariner-operator/54)

### CODE REVIEW

File: pkg/subctl/cmd/join.go [hunk 1]
```diff
@@ -69,6 +61,51 @@ var joinCmd = &cobra.Command{
 
 func joinSubmarinerCluster(subctlData *datafile.SubctlData) {
 
+	// Missing information
+	var qs = []*survey.Question{}
+
+	if len(clusterID) == 0 {
```
Comment (mangelajo) [inline pkg/subctl/cmd/join.go]: can we change it to...  len(clusterID)==0 or !isValidClusterID(str)  ?

In such way we avoid the final panic condition if the cmdline was wrong.

File: pkg/subctl/cmd/join.go [hunk 2]
```diff
@@ -69,6 +61,51 @@ var joinCmd = &cobra.Command{
...
+			Validate: func(val interface{}) error {
+				str, ok := val.(string)
+				if !ok {
+					return nil
+				}
+				return isValidClusterID(str)
+			},
+		})
+	}
+	if len(colorCodes) == 0 {
```
Comment (mangelajo) [inline pkg/subctl/cmd/join.go]: can we also add a default color in the cobra cmdline variable?, that way we avoid asking the admin for this field. 

This is something rancher internal I think we even talked to chris once about removal... so i'd better not to ask the admin for unnecessary stuff. They could use labels/annotations on the clusters for that.

File: pkg/subctl/cmd/join.go [hunk 3]
```diff
@@ -69,6 +61,51 @@ var joinCmd = &cobra.Command{
...
+		if len(answers.ClusterID) > 0 {
+			clusterID = answers.ClusterID
+		}
+		if len(answers.ColorCodes) > 0 {
+			colorCodes = answers.ColorCodes
+		}
+	}
+
+	// Make sure the clusterid is valid (we check twice because this might come from the command line)
+	err := isValidClusterID(clusterID)
```
Comment (mangelajo) [inline pkg/subctl/cmd/join.go]: can be removed with the top if condition change.

File: pkg/subctl/cmd/join.go (hunk 1 above)
Comment (skitt) [reply pkg/subctl/cmd/join.go]: Ah yes, I like that.

File: pkg/subctl/cmd/join.go (hunk 2 above)
Comment (skitt) [reply pkg/subctl/cmd/join.go]: Yes, indeed.

Comment (mangelajo) [reply pkg/subctl/cmd/join.go]: ack thanks !! :)

Comment (skitt) [reply pkg/subctl/cmd/join.go]: I’ve left the colour question here, but with #55 it won’t be asked unless the user explicitly specifies a blank colour code on the command line.

**Pull request merged**